app/models/annotation/nml/NmlParser.scala

```diff
@@ -379,8 +380,10 @@ class NmlParser @Inject()(datasetDAO: DatasetDAO) extends LazyLogging with Proto
   private def parseWkUrl(nodes: NodeSeq): Option[String] =
     nodes.headOption.map(node => getSingleAttribute(node, "wkUrl"))
 
-  private def parseOrganizationId(nodes: NodeSeq): String =
-    nodes.headOption.map(node => getSingleAttribute(node, "organization")).getOrElse("")
+  private def parseOrganizationId(nodes: NodeSeq, fallbackOrganization: String): String =
+    nodes.headOption
+      .map(node => getSingleAttributeOpt(node, "organization").getOrElse(fallbackOrganization))
```

Comment:
```suggestion
      .flatMap(node => getSingleAttributeOpt(node, "organization"))
```
would a flstMap work here?

Reply:
Yeah, good suggestion. Should work :tada: